libs/providers/flagd/src/lib/flagd-provider.ts

```diff
@@ -8,24 +8,38 @@ export class FlagdProvider implements Provider {
     name: 'flagd Provider',
   };
 
-  private readonly _service: Service;
+  get status() {
+    return this._status;
+  }
 
-  /**
-   * Promise indicating the gRPC stream is connected.
-   *
-   * Can be used in instances where the provider being connected to the event stream is a prerequisite
-   * to execution (e.g. testing). Not necessary for standard usage.
-   *
-   * @returns true if stream connected successfully, false if connection not enabled.
-   */
-  get streamConnection() {
-    return this._service.streamConnection;
+  get events() {
+    return this._events;
   }
 
+  private readonly _service: Service;
+  private _status = ProviderStatus.NOT_READY; 
+  private _events = new OpenFeatureEventEmitter();
+
   constructor(options?: FlagdProviderOptions, service?: Service, private logger?: Logger) {
     this._service = service ? service : new GRPCService(getConfig(options), undefined, logger);
   }
 
+  initialize(): Promise<void> {
```

Comment:
@beeme1mr the `service` argument here is only used for testing. I think we should move this to the last arg, and move the logger up, since this is already a breaking change.

I will also add some in-line doc.